### ISSUE TITLE
🐛 [flutter_ui] Sign in error message causes UI overflow

### ISSUE THREAD

**Issue by elvisun** [firebase/flutterfire#8396]
## Bug report

**Describe the bug**
Sign in error message causes UI overflow on `flutterfire_ui: ^0.4.0`

### Steps to reproduce

1. Create a basic SignInScreen widget
2. Run app and produce error
3. See error causes ui to overflow

```
SignInScreen(
      providerConfigs: [
        EmailProviderConfiguration(),
        GoogleProviderConfiguration(),
      ],
    );
```

![screencapture-localhost-5000-2022-04-04-11_14_51](https://user-images.githubusercontent.com/6955348/161576471-20f84137-232f-4e06-b205-daaed1593ede.png)

**Comment by darshankawar** [comment]
@elvisun 
What device / platform are you seeing this on ? I tried the same on iPhone simulator, but don't see the overflow exception. Maybe this happens on smaller size devices.

**Comment by elvisun** [comment]
Was on web, chrome desktop.

```
[√] Flutter (Channel stable, 2.10.3, on Microsoft Windows [Version 10.0.19043.1586], locale en-US)
    • Flutter version 2.10.3 at D:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (5 weeks ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\1\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Build Tools 2019 16.5.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools
    • Visual Studio Build Tools 2019 version 16.5.30002.166
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 51.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Community Edition (version 2017.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2017.2.3
    • Flutter plugin version 21.2.2
    • Dart plugin version 172.3968.27

[√] Connected device (4 available)
    • Pixel 3a (mobile) • 04FAY1W0RT • android-arm64  • Android 10 (API 29)
    • Windows (desktop) • windows    • windows-x64    • Microsoft Windows [Version 10.0.19043.1586]
    • Chrome (web)      • chrome     • web-javascript • Google Chrome 99.0.4844.84
    • Edge (web)        • edge       • web-javascript • Microsoft Edge 99.0.1150.46

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by darshankawar** [comment]
Thanks for the update. I see the same behavior using plugin example and running on web and desktop respectively.
I removed the `footerBuilder` and had only `EmailProviderConfiguration()`.

![161576471-20f84137-232f-4e06-b205-daaed1593ede](https://user-images.githubusercontent.com/67046386/161953820-7e26832b-9cdd-4557-8894-9a73a6069b95.png)

`Channel stable, 2.10.4`

**Comment by lesnitsky** [comment]
fixed in #9343
